Snowflake.Data/Core/FileTransfer/StorageClient/SFGCSClient.cs

```diff
@@ -263,11 +207,15 @@ public async Task<FileHeader> GetFileHeaderAsync(SFFileMetadata fileMetadata, Ca
                         contentLength = Convert.ToInt64(contentLength[0])
                     };
                 }
-                catch (WebException ex)
-                {
-                    fileMetadata = HandleFileHeaderErr(ex, fileMetadata);
-                }
             }
+            catch (WebException ex)
+            {
+                // GCS presigned urls have a different error handling for file headers
```

Comment:
let's log it as well

Reply:
Added the logging in the error handling functions, but if you prefer it logged here instead let me know and I'll change it